server/athenian/api/internal/prefixer.py

```diff
@@ -298,3 +299,20 @@ def __repr__(self) -> str:
     def __sentry_repr__(self) -> str:
         """Override {}.__repr__() in Sentry."""
         return repr(self)
+
+
+class LazyPrefixerProxy:
+    """When called return an instance of Prefixer, building it only when needed."""
+
+    def __init__(self, request, account):
+        """Init the `LazyPrefixerProxy`."""
+
+        @alru_cache
+        async def get_prefixer():
+            return await Prefixer.from_request(request, account)
```

Comment:
Finally, this is likely not working as you'd expect. When you return an async function, it doesn't actually start to execute! You must wrap it into `asyncio.Task` to let it execute along with the rest of the code.

Also, we already have the memcached. It takes very few milliseconds to load the cached prefixed. So we won't win much here.

Reply:
:thinking: yes I don't want to execute it, it's lazy after all